### PR TITLE
refactor(console, core): refactor console to support new pro plan

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR refactors the console and core code to support our new Pro plan migration. 

### Refactord
1. Add some missing comments to make the context clear
2. Refactor the `PlanUsage` component. Make the `periodicUsage` props required. 
   - For `MauExceededModel` we should use the usage data from the tenant context.
   - For `Subscription/CurrentPlan` we should always fetch the latest usage data.
3. Extract the SKU sorting logic from `formatLogtoSkusResponses` method to the  `pickupFeaturedLogtoSkus` method. We only need those SKUs to be ordered in the subscription plan table. No need to sort the data on every Logto SKU request. 
4. Extract a new `SkuButton` child component, to avoid complicated inline element display logic. 
   
 ### New Pro plan support
1. Update the `featuredPlanIds` list. Once the new plan is launched, the latest `Pro202411` ID should replace the current `Pro` ID.
2. Refactor the `featuredPlanIdOrder` to a map instead of a list. So we can support multiple plans that share the same order, e.g grandfathered `Pro` and `Pro202411`.
3. Update the `PlanComparisonTable` content to pick up the latest quota updates.
4. Introduce a new `isEquivalentPlan` util method. If the given planId is not equal to the target planId, but they both share the same plan order, we treat them as equivalent plans. New  `Pro202411` and grandfather `Pro`.  We display a Contact Us button for such a case. 
image

## Testing

test locally

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/components/MauExceededModal/index.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { cond } from '@silverhand/essentials';
-import { useContext, useState } from 'react';
+import { cond, conditional } from '@silverhand/essentials';
```
Comment (xiaoyijun) [inline packages/console/src/components/MauExceededModal/index.tsx:1]: `cond` and `conditional` are the same.

File: packages/console/src/components/PlanUsage/index.tsx [hunk 2]
```diff
@@ -26,7 +25,7 @@ import {
 } from './utils';
 
 type Props = {
-  readonly periodicUsage?: NewSubscriptionPeriodicUsage;
+  readonly periodicUsage: NewSubscriptionPeriodicUsage | undefined;
```
Comment (xiaoyijun) [inline packages/console/src/components/PlanUsage/index.tsx:28]: Is this change necessary?

File: packages/console/src/consts/subscriptions.ts [hunk 3]
```diff
@@ -25,15 +25,25 @@ export const tokenAddOnUnitPrice = 80;
 export const hooksAddOnUnitPrice = 2;
 /* === Add-on unit price (in USD) === */
 
+// TODO: Remove this dev feature flag when we have the new Pro202411 plan released.
```
Comment (xiaoyijun) [inline packages/console/src/consts/subscriptions.ts:28]: Maybe add an owner for this TODO.

**Review by xiaoyijun — APPROVED** [review]
Overall LGTM.

File: packages/console/src/components/PlanUsage/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/components/PlanUsage/index.tsx:28]: Yes, it indicates that `periodicUsage` property is required when using this component, but the value can be undefined.

File: packages/console/src/consts/subscriptions.ts (hunk 3 above)
Comment (simeng-li) [reply packages/console/src/consts/subscriptions.ts:28]: No need will query the `isDevFeaturesEnabled` flag.

**Review by darcyYe — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]
